src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3211,3 +3224,236 @@ def _aks_approuting_update(
+                    managed_resource_group, vmss_name, instance_id,
+                    RunCommandInput(command_id="RunShellScript", script=[command])))
+        elif vm_set_type == CONST_AVAILABILITY_SET:
+            RunCommandInput = cmd.get_models('RunCommandInput',
+                                             resource_type=ResourceType.MGMT_COMPUTE,
+                                             operation_group="virtual_machine_run_commands")
+            command_result = LongRunningOperation(cmd.cli_ctx)(
+                compute_client.virtual_machines.begin_run_command(
+                    managed_resource_group, vm_name,
+                    RunCommandInput(command_id="RunShellScript", script=[command])))
+        else:
+            raise ValidationError(f"VM set type {vm_set_type} is not supported!")
+
+        display_status = command_result.value[0].display_status
+        message = command_result.value[0].message
+        if display_status != "Provisioning succeeded":
+            raise InvalidArgumentValueError(
+                f"Can not run command with returned code {display_status} and message {message}")
+        return process_message_for_run_command(message)
+    except Exception as ex:
+        raise HttpResponseError(f"Can not run command with returned exception {ex}") from ex
+
+
+def _aks_verify_resource(resource, resource_type):
+    if resource.provisioning_state != CONST_NODE_PROVISIONING_STATE_SUCCEEDED:
+        raise ValidationError(f"Node pool {resource.name} is in {resource.provisioning_state} state!")
+
+    node_image_version = ""
+    os_type = ""
+    if resource_type == CONST_VIRTUAL_MACHINE_SCALE_SETS:
+        node_image_version = resource.node_image_version
+        os_type = resource.os_type
+    else:
+        node_image_version = resource.storage_profile.image_reference.id
+        os_type = resource.storage_profile.os_disk.os_type
+
+    if not os_type or os_type != CONST_DEFAULT_NODE_OS_TYPE:
+        raise ValidationError(f"Resource must be of type {CONST_DEFAULT_NODE_OS_TYPE}!")
+
+    if not node_image_version:
+        raise ValidationError(f"No image version found for {resource.name}! Cannot verify supported versions.")
+
+    if resource_type == CONST_VIRTUAL_MACHINE_SCALE_SETS:
+        version = node_image_version.split("-")[-1]
+    else:
+        version = node_image_version.split("/")[-1]
+
+    if version < CONST_MIN_NODE_IMAGE_VERSION:
+        raise ValidationError(f"Node image version {version} is not supported! "
+                              f"Image version must be at least {CONST_MIN_NODE_IMAGE_VERSION}.")
+
+
+def _aks_get_node_name_vmss(
+        cmd,
+        resource_group,
+        cluster_name,
+        node_name,
+        managed_resource_group):
+    compute_client = get_compute_client(cmd.cli_ctx)
+
+    if not node_name:
+        print("No node name specified, will randomly select a node from the cluster")
```

Comment:
looks like a warning message, consider using `logger.warning`

BTW, print will be output to stdout and mixed with the return value of the command., while logger.warning will be output to stderr.

Reply:
Yes, I fully understand that print will be output to stdout as it was my intention. For this command, --node-name is optional so `No node name specified` isn't really a warning. For this, I guess I can probably use logger.debug here.